src/query/api/v1/handler/prometheus/native/common.go

```diff
@@ -75,9 +78,14 @@ func parseDuration(r *http.Request, key string) (time.Duration, error) {
 		return value, nil
 	}
 
-	// Try parsing as an integer value specifying seconds, the Prometheus default
-	if seconds, intErr := strconv.ParseInt(str, 10, 64); intErr == nil {
-		return time.Duration(seconds) * time.Second, nil
+	// Try parsing as a float value specifying seconds, the Prometheus default
+	if seconds, floatErr := strconv.ParseFloat(str, 64); floatErr == nil {
+		ts := seconds * float64(time.Second)
+		if ts > maxInt64 || ts < minInt64 {
```

Comment:
Do you need to validate for negative step size in Prom? If so, not sure if you should do it here or somewhere else.

Reply:
Cool - just added to the parseParams() function - similar to what prom does.